src/components/tree/Tree.tsx

```diff
@@ -0,0 +1,152 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import { useDebounce } from 'react-use';
+
+import { useFilters } from '@/hooks/useFilters';
+import { useToolbox } from '@/hooks/useToolbox';
+import { Collection, Link } from '@/typings/stac';
+
+import { TreeNode } from './TreeNode';
+import { filterTree } from './utils';
+
+import './Tree.scss';
+
+export type TreeCatalog = {
+  id: string;
+  title: string;
+  type: 'Catalog';
+  catalogs?: TreeCatalog[]; // sub-catalogs
+  collections?: Collection[]; // collections
+};
+
+// Utility function to fetch data from a given URL
+const fetchData = async (url: string) => {
+  const response = await fetch(url);
+  return response.json();
+};
+
+type TreeProps = {
+  catalogUrl: string;
+};
+
+export const Tree = ({ catalogUrl }: TreeProps) => {
+  const [treeData, setTreeData] = useState<TreeCatalog[]>([]);
+  const [expandedNodes, setExpandedNodes] = useState<{ [key: string]: boolean }>({});
+  const [filteredTreeData, setFilteredTreeData] = useState<TreeCatalog[]>([]);
+
+  const {
+    state: { activeFilters },
+  } = useFilters();
+
+  const {
+    actions: { setActivePage, setSelectedCollection },
+  } = useToolbox();
+
+  // Recursive function to fetch and build the tree structure
+  const fetchCatalog = useCallback(async (url: string): Promise<TreeCatalog> => {
```

Comment:
This feels like a function that should be in a context somewhere, it seems likely that another component may need to rebuild the tree

Reply:
I thought of this but didn't want to over-engineer the solution either, so I decided that if another component did need it, then was the right time to move it out into context.

Right now with tight deadline for this sprint, I would personally prefer a new ticket to do that, if and when it was necessary, what do you think? If I do it now, I'm concerned the whole thing will have to slip into the next sprint and I am expected to go down to half time very soon and then onto another project in November.